sample/src/main/java/com/google/accompanist/sample/permissions/RequestPermissionSample.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.sample.permissions
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.provider.Settings
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.rememberPermissionState
+import com.google.accompanist.sample.AccompanistSampleTheme
+
+class RequestPermissionSample : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            AccompanistSampleTheme {
+                val cameraPermissionState = activityResultRegistry.rememberPermissionState(
+                    android.Manifest.permission.CAMERA,
+                )
+                Sample(
+                    cameraPermissionState,
+                    navigateToSettingsScreen = {
+                        startActivity(
+                            Intent(
+                                Settings.ACTION_APPLICATION_DETAILS_SETTINGS,
+                                Uri.fromParts("package", packageName, null)
+                            )
+                        )
+                    }
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun Sample(
+    cameraPermissionState: PermissionState,
+    navigateToSettingsScreen: () -> Unit
+) {
+    when {
+        cameraPermissionState.hasPermission -> {
+            Text("Camera permission Granted")
+        }
+        cameraPermissionState.shouldShowRationale -> {
+            Column {
+                Text("The camera is important for this app. Please grant the permission.")
+                Spacer(modifier = Modifier.height(8.dp))
+                Button(onClick = { cameraPermissionState.launchPermissionRequest() }) {
```

Comment:
Might be worth to showcase the best practices here, by offering a way to cancel the rational. That would go to a similar state in the else clause.

This is not part of the API but worth adding it.

Reply:
Done! Thanks